src/index.ts

```diff
@@ -13,80 +13,26 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import * as assert from 'assert';
-import * as fs from 'fs';
-import * as stream from 'stream';
-import * as util from 'util';
-import * as zlib from 'zlib';
 
-import {serialize} from './builder';
-import {perftools} from './profile';
+import {ProfileAgentConfig} from './config';
+import {ProfileAgent} from './agent';
 
-type HrTimeTuple = [number, number];
-
-const profiler = require('bindings')('cpu_profiler');
-
-function timeToNanos(tuple: HrTimeTuple) {
-  return tuple[0] * 1e9 + tuple[1];
+/**
+ * Start the profiling agent.
```

Comment:
"Starts"

Also:
* Can it fail? If so, how errors are signaled?
* Can it be stopped?